src/js/repl.js

```diff
@@ -209,19 +216,31 @@ function highlight(rlInt: readline$Interface,
   }
 }
 
-function handleKeyPress(rl: readline$Interface, c: string, key: KeyType): void {
+function handleKeyPress(sessionId: number, rl: readline$Interface, c: string, key: KeyType): void {
   const now = currentTimeMicros();
   isPasting = (now - lastKeypressTime) < 10000;
   lastKeypressTime = now;
 
   if (!isPasting) {
     // $FlowIssue: these properties exist
-    highlight(rl, c, rl.line, rl.cursor);
+    highlight(sessionId, rl, c, rl.line, rl.cursor);
   }
 }
 
+export function createSession(): number {
+  sessionCount += 1;
+  const sessionId = sessionCount - 1; // start at 0
```

Comment:
this arithmetic would be pointless to make if we stored the sessions in an object.